src/main/java/com/iridium/iridiumskyblock/api/UserChatToggleEvent.java

```diff
@@ -0,0 +1,43 @@
+package com.iridium.iridiumskyblock.api;
+
+import com.iridium.iridiumskyblock.database.Island;
+import com.iridium.iridiumskyblock.database.User;
+import lombok.Getter;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+@Getter
+public class UserChatToggleEvent extends Event implements Cancellable {
+
+    private static final HandlerList handlers = new HandlerList();
+    private boolean cancelled;
+    private final Island island;
+    private final User user;
+
+    public UserChatToggleEvent(Island island, User user) {
```

Comment:
This should also contain a state, like boolean chatEnabled, ye I know they can do user.getIslandChat or whatever but I feel like it should be exposed here aswell to make it easier